packages/taquito-beacon-wallet/test/taquito-beacon-wallet.spec.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  BeaconWallet,
+  BeaconWalletNotInitialized,
+  MissingRequiredScopes,
+} from '../src/taquito-beacon-wallet';
+import LocalStorageMock from './mock-local-storage';
+import { PermissionScope, LocalStorage, windowRef } from '@airgap/beacon-dapp';
+import sinon from 'sinon';
+
+global.localStorage = new LocalStorageMock();
+
+describe('Beacon Wallet tests', () => {
+
+  beforeEach(() => {
+    sinon.restore()
+      ; (windowRef as any).beaconCreatedClientInstance = false
+  })
+
+  it('Verify that BeaconWallet is instantiable', () => {
+    expect(new BeaconWallet({ name: 'testWallet' })).toBeInstanceOf(
+      BeaconWallet
+    );
+  });
+
+  it('Verify BeaconWallet not initialized error', () => {
+    expect(new BeaconWalletNotInitialized()).toBeInstanceOf(Error);
+  });
+
+  it('Verify BeaconWallet permissions scopes not granted error', () => {
+    expect(
+      new MissingRequiredScopes([PermissionScope.OPERATION_REQUEST])
+    ).toBeInstanceOf(Error);
+  });
+
+  it('Verify that permissions must be called before getPKH', async () => {
+    try {
+      const wallet = new BeaconWallet({ name: 'testWallet' });
+      await wallet.getPKH();
+    } catch (error: any) {
+      expect(error.message).toContain(
+        'You need to initialize BeaconWallet by calling beaconWallet.requestPermissions first'
+      );
+    }
+  });
+
+  it(`Verify that a Beacon Wallet has a beacon ID`, async () => {
+    const wallet = new BeaconWallet({ name: 'Test', storage: new LocalStorage() })
+    expect(typeof (await wallet.client.beaconId)).toEqual('string')
+    expect((await wallet.client.beaconId)).toBeDefined
+  })
+
+  it(`Verify requestPermissions`, async () => {
```

Comment:
Can you explain the added value of this test?

Reply:
It adds no additional coverage so I have removed it.